docs/sphinx/releases/whats_new_1_5_0.rst

```diff
@@ -8,17 +9,542 @@
+  pools (:ref:`using the resource partitioner from hpx 1.5.0 onwards
+  <_using_rp_from_1_5_0>`).
+* A polymorphic executor has been implemented to reduce compile times as a
+  function accepting executor can potentially be instanciated only once instead
+  of multiple times with different executors. It accepts the function signature
+  as a template argument. It needs to be constructed from any other executor.
+  Please note, that the function signatures that can be scheduled using
+  ``then_execute``, ``bulk_sync_execute``, ``bulk_async_execute`` and
+  ``bulk_then_execute`` are slightly different.
+* The underlying executor of ``block_executor`` has been updated to a newer one.
+* We have added a parameter to ``auto_chunk_size`` to control the amount of
+  iterations to measure.
+* We have added performance counters type information (aggregating,
+  monotonically increasing, average count, average timer, etc.).
+* The tasks are now resumed on the worker thread they were suspended to avoid
+  cache misses from moving from one thread to the other.
+* We have added a new configuration option ``hpx.exception_verbosity`` to allow
+  to control the level of verbosity of the exceptions (3 levels available).
+* ``broadcast_to``, ``broadcast_from``, ``scatter_to`` and ``scatter_from`` have
+  been added to the collectives, modernization of ``gather_here`` and
+  ``gather_there`` with futures taken by rvalue references, rename the existing
+  ``all_to_all`` to ``all_gather`` and adds a new implementation of
+  ``all_to_all``.
+* A new API function to get the number of cores which are idle
+  (``hpx::get_idle_core_count``)
+* We have added an experimental option to only enable the local runtime, you can
+  enable the distributed runtime with ``HPX_WITH_DISTRIBUTED_RUNTIME=ON`` (or
+  ``OFF`` to disable it and have only the local one). You can also enable the
+  local runtime by using the ``--hpx:local`` runtime option.
+* We fixed task annotations for actions.
+* ``hpx::promise`` is now deprecated and transformed into ``hpx::lcos::promise``.
+* ``result_of`` and ``is_callable`` are now deprecated and replaced by
+  ``invoke_result``.
+* We have added a ``prepare_checkpoint`` API function which calculates the
+  amount of necessary buffer space for a particular set of argumnets
+  checkpointed.
+* We have changed the exposed targets to the user, it now includes ``HPX::hpx``,
+ ``HPX::wrap_main`` (``int main`` as the first HPX::thread of the application,
+ see :ref:`starting HPX runtime <_starting_hpx>`),
+ ``HPX::plugin``, ``HPX::component``.
+* A new example is demonstrating how to create and use a wrapping executor
+  (``quickstart/executor_with_thread_hooks.cpp``)
+* A new example is demonstrating how to disable thread stealing during the
+  execution of parallel algorithms
+  (``quickstart/disable_thread_stealing_executor.cpp``)
+* Many bug fixes.
+
+Breaking changes
+================
+
+* HPX now requires C++14 capable compiler.
```

Comment:
also:

- runtime initialization using `resource_partitioner`
- `all_to_all` is now called `all_gather`, new `all_to_all` is not compatible with old